rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/KubernetesApiImpl.scala

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright (C) 2017-2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.rollingupdate.kubernetes
+
+import scala.collection.immutable
+import scala.concurrent.Future
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.HttpsConnectionContext
+import akka.http.scaladsl.marshalling.Marshal
+import akka.http.scaladsl.model.HttpEntity
+import akka.http.scaladsl.model.HttpMethods.PATCH
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.model.MediaTypes
+import akka.http.scaladsl.model.StatusCodes.ClientError
+import akka.http.scaladsl.model.Uri
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.Authorization
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.pattern.after
+import akka.util.ByteString
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] class KubernetesApiImpl(
+    system: ActorSystem,
+    settings: KubernetesSettings,
+    override val namespace: String,
+    apiToken: String,
+    clientHttpsConnectionContext: Option[HttpsConnectionContext])
+    extends KubernetesApi
+    with KubernetesJsonSupport {
+
+  import system.dispatcher
+
+  private implicit val sys: ActorSystem = system
+  // FIXME the asInstanceOf is because Scala3 complains
+  private val log = Logging(system, getClass.asInstanceOf[Class[Any]])
```

Comment:
In Akka we do `Logging(system, classOf[ThisClass])` to make Scala 3 happy

Reply:
ah, that seems obvious and better. I recognize this hack and I could have been guilty of introducing it at the first place and then it has been copied to many places. @sebastian-alfers would you be able to look for this pattern in the various repos and PR Johan's suggestion?